application-analytics-ui/src/main/resources/Analytics/Code/Macros/MostViewedPagesJSON.xml

```diff
@@ -62,10 +62,19 @@
   })
   #set ($sort = 'nb_hits')
   #set ($order = 'desc')
-  #if ($request.sort &amp;&amp; $request.reqNo!='1')
-    #set($sort = $keys.get($request.sort))
-    #set($order = $request.dir)
+  #if ($request.sort)
+    #set ($sort = $keys.get($request.sort))
+    #set ($order = $request.dir)
+  #end
+
+  #set ($filters={})
+  #foreach ($key in $keys.keySet())
```

Comment:
I'll add a comment for the `keys` usage. Maybe also use a more specific name `matomoKeysMapping`, or just `keysMapping`?